### ISSUE TITLE
Optional Time Step Units Required

### ISSUE THREAD

**Issue by william-silversmith** [jhuapl-boss/intern#13]
| intern version |  os  | python |
|-----|------|-------|
| 0.9.4   |  OS X Sierra    | 2.7     |

When attempting to fetch an experiment using `rmt.get_project(expr)`, I got the following error:

```
>>> x = rmt.get_project(e)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/wms/neuroglancer/python/venv/lib/python2.7/site-packages/intern/remote/boss/remote.py", line 719, in get_project
    return self.project_service.get(resource)
  File "/Users/wms/neuroglancer/python/venv/lib/python2.7/site-packages/intern/service/boss/project.py", line 434, in get
    self.session_send_opts)
  File "/Users/wms/neuroglancer/python/venv/lib/python2.7/site-packages/intern/service/boss/v1/project.py", line 790, in get
    return self._create_resource_from_dict(resource, resp.json())
  File "/Users/wms/neuroglancer/python/venv/lib/python2.7/site-packages/intern/service/boss/v1/project.py", line 972, in _create_resource_from_dict
    return self._get_experiment(dict, resource.coll_name)
  File "/Users/wms/neuroglancer/python/venv/lib/python2.7/site-packages/intern/service/boss/v1/project.py", line 998, in _get_experiment
    collection_name=coll_name, raw=dict, **filtered)
  File "/Users/wms/neuroglancer/python/venv/lib/python2.7/site-packages/intern/resource/boss/resource.py", line 204, in __init__
    self._time_step_unit = self.validate_time_units(time_step_unit)
  File "/Users/wms/neuroglancer/python/venv/lib/python2.7/site-packages/intern/resource/boss/resource.py", line 270, in validate_time_units
    raise ValueError('{} is not a valid time unit.'.format(value))
ValueError:  is not a valid time unit.
```

This was resolved when I entered some bogus time information in the appropriate fields. 

I think this is the latest version on PyPI. Sorry if this is already fixed.

Will

**Comment by movestill** [comment]
Will, this was fixed in commit 8b7ebfe. Unfortunately, we've been understaffed lately, so we haven't had a chance to update the PyPI version.

**Comment by william-silversmith** [comment]
Thanks Tim!

On Aug 7, 2017 12:28 PM, "Tim Gion" <notifications@github.com> wrote:

Will, this was fixed in commit 8b7ebfe
<https://github.com/jhuapl-boss/intern/commit/8b7ebfefc8cb716bbb21a6dd2223a0eea6b97189>.
Unfortunately, we've been understaffed lately, so we haven't had a chance
to update the PyPI version.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/jhuapl-boss/intern/issues/13#issuecomment-320712220>,
or mute the thread
